privacyidea/lib/user.py

```diff
@@ -143,8 +143,6 @@ def __str__(self):
         ret = u"<empty user>"
         if not self.is_empty():
             login = self.login
-            if not isinstance(login, six.text_type):
```

Comment:
Can we be sure that ``login`` is always ``unicode`` in Python 2 resp. ``str`` in Python 3? It definitely should be, but I'm not sure if it's really the case currently :-)
But still, I think it is reasonable to remove these checks in 3.0. What do you think about adding a warning at the end of ``User.__init__`` like
```python
if not isinstance(login, six.text_type):
    log.warning("Unexpected bytestring as login name: {!r}".format(login))
```
?

Reply:
I think we should be on the safe side, even if it is a `str` type in Python 2. It is mostly used in logging and deleting the user cache. These should work just fine.
And `self.login` is only written in 3 places so if we run into trouble, there is a quick fix :-)

Reply:
I guess you're right -- let's do it that way :)